editor/editor_command_palette.cpp

```diff
@@ -0,0 +1,254 @@
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "editor/editor_command_palette.h"
+#include "core/os/keyboard.h"
+#include "editor/editor_node.h"
+#include "scene/gui/tree.h"
+
+EditorCommandPalette *EditorCommandPalette::singleton = nullptr;
+
+void EditorCommandPalette::_update_command_search() {
+	List<String> *actions_list = memnew(List<String>);
+	get_actions_list(actions_list);
+
+	if (actions_list != nullptr) {
+		const String search_text = command_search_box->get_text();
+		const bool empty_search = search_text == "";
+
+		// Filter possible candidates.
+		Vector<Pair<String, String>> entries;
+		for (int i = 0; i < actions_list->size(); i++) {
+			String key_name = (*actions_list)[i];
+			String display_name = _get_command_name(key_name);
+			if (!empty_search && search_text.is_subsequence_ofi(display_name)) {
+				// To-do: Rank the Entries....
+				// r.score = empty_search ? 0 : _score_path(search_text, files[i].to_lower());
+				// print_line("found entry : " + (*actions_list)[i]);
+				Pair entry = Pair(key_name, display_name);
+				entries.push_back(entry);
+			}
+		}
+
+		actions_list->clear();
+
+		TreeItem *root = search_options->get_root();
+		root->clear_children();
+
+		if (entries.size() > 0) {
+			// if (!empty_search) {
+			// 	SortArray<Entry, EntryComparator> sorter;
+			// 	sorter.sort(entries.ptrw(), entries.size());
+			// }
+
+			const int entry_limit = MIN(entries.size(), 300);
+			for (int i = 0; i < entry_limit; i++) {
+				TreeItem *ti = search_options->create_item(root);
+				ti->set_tooltip(0, entries[i].first);
+				ti->set_text(0, entries[i].second);
+				// ti->set_icon(0, *icons.lookup_ptr(entries[i].path.get_extension()));
+			}
+
+			TreeItem *to_select = root->get_first_child();
+			to_select->select(0);
+			to_select->set_as_cursor(0);
+			search_options->scroll_to_item(to_select);
+
+			get_ok_button()->set_disabled(false);
+		} else {
+			search_options->deselect_all();
+
+			get_ok_button()->set_disabled(true);
+		}
+	} else {
+		print_error("Failed to fetch action_list.");
+	}
+}
+
+void EditorCommandPalette::_text_changed(const String &p_newtext) {
```

Comment:
Instead of creating a function only for that, you could connect the LineEdit signal like that:
`command_search_box->connect("text_changed", callable_mp(this, &EditorCommandPalette::_update_command_search).unbind(1));`

Reply:
yep